controllers/providers/assistant/gslb.go

```diff
@@ -60,27 +61,42 @@ func NewGslbAssistant(client client.Client, k8gbNamespace, edgeDNSServer string)
 
 // CoreDNSExposedIPs retrieves list of IP's exposed by CoreDNS
 func (r *GslbLoggerAssistant) CoreDNSExposedIPs() ([]string, error) {
+	serviceList := &corev1.ServiceList{}
 	coreDNSService := &corev1.Service{}
-	err := r.client.Get(context.TODO(),
-		types.NamespacedName{Namespace: r.k8gbNamespace, Name: coreDNSExtServiceName}, coreDNSService)
+	sel, err := labels.Parse(coreDNSServiceLabel)
+	if err != nil {
+		log.Err(err)
+	}
+	listOption := &client.ListOptions{
+		LabelSelector: sel,
+		Namespace:     r.k8gbNamespace,
+	}
+
+	err = r.client.List(context.TODO(), serviceList, listOption)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			log.Warn().Msgf("Can't find %s service", coreDNSExtServiceName)
+			log.Warn().Msgf("Can't find service %s", err.Error())
 		}
+	}
+	if len(serviceList.Items) != 1 {
+		log.Info().Msgf("items are not 1 actual %d", len(serviceList.Items))
+		err := coreerrors.New("Can't find CoreDNS service")
 		return nil, err
 	}
+	coreDNSService = &serviceList.Items[0]
+
 	var lbHostname string
 	if len(coreDNSService.Status.LoadBalancer.Ingress) > 0 {
 		lbHostname = coreDNSService.Status.LoadBalancer.Ingress[0].Hostname
 	} else {
-		errMessage := fmt.Sprintf("no Ingress LoadBalancer entries found for %s serice", coreDNSExtServiceName)
+		errMessage := fmt.Sprintf("no LoadBalancer ExternalIPs are found for %s service", coreDNSService.Name)
 		log.Warn().Msg(errMessage)
```

Comment:
```suggestion
		log.Warn().Str("CoreDNSServiceName", coreDNSService.Name).
		Msg("No LoadBalancer ExternalIPs are found for service")
```

Reply:
I see that spelling was corrected, but suggested contextual logging format (https://github.com/rs/zerolog#contextual-logging) is not being used.